### ISSUE TITLE
Improved Logging (and logging to syslog) needed

### ISSUE THREAD

**Issue by aio-witc** [Nitrokey/nethsm-pkcs11#153]
Context:
We're using the PKCS#11 module (and the NetHSM) to protect our ssh keys (for automation and for admin access). We usually add the PKCS#11 module to the running ssh-agent with ssh-add -s ... . Problem is: Text written to stderr (or stdout) is lost. Furthermore, we had issues with correct file permissions to the config file or the log file. In these cases, the PKCS#11 module fails without any visible error. Locating the underlying problem is quite difficult in these cases.

Request:
- Logging to syslog should be implemented as well. Syslog should be default (so that issues during startup are logged). The usual options should be configurable, e.g. facility (default: 1 user-level messages), actually all the RFC 5424 fields (https://www.rfc-editor.org/rfc/rfc5424.html#section-6). See "man logger" for useful options.
- Logging to different log sockets is possible. (low priority)
- Logging to remote servers is possible, even using TCP. (very low priority)

**Comment by daringer** [comment]
The first step we go is to also log to `syslog` on init-fail, roughly here: https://github.com/Nitrokey/nethsm-pkcs11/blob/ed449b6e6b348198e8b2cf36758e215545432833/pkcs11/src/data.rs#L21

Same applies to the log-file init here:
https://github.com/Nitrokey/nethsm-pkcs11/blob/ed449b6e6b348198e8b2cf36758e215545432833/pkcs11/src/config/initialization.rs#L23

These are the measures to avoid a panic i.e. crash without knowing why. Afterwards also the possibility to configure logging into syslog in general shall be added.

**Comment by aio-witc** [comment]
see: https://github.com/Nitrokey/nethsm-pkcs11/issues/161

Do not call panic! for trivial run time errors....

**Comment by daringer** [comment]
Yes, good point, means a configuration reading or log-file-based error will anyways write to syslog and then return an error.

**Comment by aio-witc** [comment]
Additional Request: Include used config file in log file, eg. "/etc/nitrokey", "/usr/local/etc/nitrokey", ~/.config/nitrokey
